### PR TITLE
use sysvinit for debian runsvdir

### PR DESCRIPTION
**Pull request by smith**
Debian defaults to sysvinit. This replaces #24. See that PR for more discussion.

/cc @chef/chef-visual-interaction

### CODE REVIEW

File: libraries/provider_component_runit_supervisor_upstart.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@ class Chef
   class Provider
     class ComponentRunitSupervisor
       class Upstart < Chef::Provider::LWRPBase
-        provides :component_runit_supervisor, :platform_family => "debian"
         provides :component_runit_supervisor, :platform_family => "rhel" do |node|
           node['platform_version'].to_i == 6
         end
-        provides :component_runit_supervisor, :platform => %w[ amazon fedora ]
+        provides :component_runit_supervisor,
+          :platform => %w[ amazon fedora ubuntu ]
```
Comment (mmzyk) [inline libraries/provider_component_runit_supervisor_upstart.rb]: This seems to be a behavior change that wasn't specified in the commit message. Was this just an oversight from when this lwrp was built? At the very least it seems like this should be in a separate commit from the debian change.

**Comment by mmzyk** [comment]
I am not a linux distro expert, or even that knowledgable about the difference between them, but given the amount of change that has been happening with init systems, does this PR take into account the different init systems debian might have across different versions? From the link PR discussion I'm not convinced we can just assume sysvinit for Debian, but I might be wrong on this and it's actually okay to do so.

File: libraries/provider_component_runit_supervisor_upstart.rb (hunk 1 above)
Comment (smith) [inline libraries/provider_component_runit_supervisor_upstart.rb]: No this just explicitly adds ubuntu where before it was being chose by process of elimination. I suppose I can make it a separate commit, but if you're ok with it I can just not do that.

**Comment by smith** [comment]
> I am not a linux distro expert, or even that knowledgable about the difference between them, but given the amount of change that has been happening with init systems, does this PR take into account the different init systems debian might have across different versions? From the link PR discussion I'm not convinced we can just assume sysvinit for Debian, but I might be wrong on this and it's actually okay to do so.

This change does what #24 proposes and from my reading of Debian's docs is correct. Since Chef Software doesn't officially support Debian in any of the projects that use this, I'd say it's safe to ship and can be corrected if things change or if assumptions are incorrect.

**Comment by mmzyk** [comment]
All the reasoning sounds good to me. While we don't officially support Debian, I like keeping our Debian friends happy. :shipit:

**Comment by gplessis** [comment]
Thank you all!
